packages/flutter/lib/src/widgets/viewport.dart

```diff
@@ -98,3 +98,38 @@ class Viewport2Element extends MultiChildRenderObjectElement {
     }
   }
 }
+
+class ShrinkWrappingViewport extends MultiChildRenderObjectWidget {
+  ShrinkWrappingViewport({
+    Key key,
+    this.axisDirection: AxisDirection.down,
+    @required this.offset,
+    List<Widget> slivers: const <Widget>[],
+  }) : super(key: key, children: slivers) {
+    assert(offset != null);
+  }
+
+  final AxisDirection axisDirection;
+  final ViewportOffset offset;
+
+  @override
+  RenderShrinkWrappingViewport createRenderObject(BuildContext context) {
+    return new RenderShrinkWrappingViewport(
+      axisDirection: axisDirection,
+      offset: offset,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, RenderShrinkWrappingViewport renderObject) {
+    renderObject.axisDirection = axisDirection;
+    renderObject.offset = offset;
```

Comment:
We've been using `..` in updateRenderObject functions because it's slightly more efficient and these functions are super hot.